ote_sdk/ote_sdk/entities/shapes/rotated_rectangle.py

```diff
@@ -0,0 +1,187 @@
+            >>> p1 = RotatedRectangle(points=[Point(x=0.5, y=0.25),
+            ... Point(x=0.75, y=0.5), Point(x=0.5, y=0.75), Point(x=0.25, y=0.5)])
+            >>> roi = Rectangle(x1=0.0, x2=0.5, y1=0.0, y2=0.5)
+            >>> normalized = p1.normalize_wrt_roi_shape(roi)
+            >>> normalized
+            RotatedRectangle(points=[Point(0.25, 0.125), Point(0.375, 0.25), Point(0.25, 0.375), Point(0.125, 0.25)])
+
+        :param roi_shape: Region of Interest
+        :return: New RotatedRectangle in the image coordinate system
+        """
+
+        if not isinstance(roi_shape, Rectangle):
+            raise ValueError("roi_shape has to be a Rectangle.")
+
+        roi_shape = roi_shape.clip_to_visible_region()
+
+        points = [p.normalize_wrt_roi(roi_shape) for p in self.points]
+        return RotatedRectangle(points=points)
+
+    def denormalize_wrt_roi_shape(self, roi_shape: Rectangle) -> "RotatedRectangle":
+        """
+        Transforming shape from the normalized coordinate system to the `roi` coordinate system.
+        This function is the inverse of ``normalize_wrt_roi_shape``
+
+        :example: Assume we have RotatedRectangle `p1` which lives in the middle of the normalized coordinate space.
+            The `roi` is a rectangle `roi` living in the top-left quarter of the normalized coordinate
+            space.
+            This function returns RotatedRectangle `p1`expressed in the coordinate space of `roi`.
+
+            RotatedRectangle denormalized to a rectangle as ROI
+
+        >>> from ote_sdk.entities.annotation import Annotation
+        >>> from ote_sdk.entities.shapes.polygon import Point
+        >>> from ote_sdk.entities.shapes.rectangle import Rectangle
+        >>> from ote_sdk.entities.shapes.rotated_rectangle import RotatedRectangle
+        >>> p1 = RotatedRectangle(points=[Point(0.25, 0.125), Point(0.375, 0.25),
+        ... Point(0.25, 0.375), Point(0.125, 0.25)])
+        >>> roi = Rectangle(x1=0.0, x2=0.5, y1=0.0, y2=0.5)
+        >>> normalized = p1.denormalize_wrt_roi_shape(roi)
+        >>> normalized
+        RotatedRectangle(points=[Point(0.5, 0.25), Point(0.75, 0.5), Point(0.5, 0.75), Point(0.25, 0.5)])
+
+        :param roi_shape: Region of Interest
+        :return: New RotatedRectangle in the ROI coordinate system
+        """
+
+        if not isinstance(roi_shape, Rectangle):
+            raise ValueError("roi_shape has to be a Rectangle.")
+
+        roi_shape = roi_shape.clip_to_visible_region()
+
+        points = [p.denormalize_wrt_roi_shape(roi_shape) for p in self.points]
+        return RotatedRectangle(points=points)
+
+    def _as_shapely_polygon(self) -> shapely_polygon:
+        """
+        Returns the RotatedRectangle object as a shapely polygon
+        which is used for calculating intersection between shapes.
+        """
+        return shapely_polygon([(point.x, point.y) for point in self.points])
```

Comment:
I think it depends on Y-axis orientation, if it points down, a polygon is cw, otherwise is ccw

I assume that our normalized coordinates are comes from image plane where Y-axis is directed downward. If my assumption is wrong, please let me know.